### PR TITLE
Fixes 4624: add template delete modal

### PR DESCRIPTION
**Pull request by jlsherrill**
## Summary

Adds a delete modal for templates when deleting a template:
* via the kabab menu in the templates list
*  via the actions drop down in the templates detail

When there is a system assigned:
image

When there are no systems assigned: 
image

### CODE REVIEW

**Comment by jlsherrill** [comment]
https://issues.redhat.com/browse/HMS-4624

**Comment by xbhouse** [comment]
nitpicky but in the delete modal, should we say instead `This template is assigned to <num> systems?`. it would be more consistent with the other language we have in the UI (`Assign template to systems`)

**Comment by xbhouse** [comment]
also can you run `npm run format` to make sure the formatting is kept consistent?

File: src/Routes/constants.ts [hunk 1]
```diff
@@ -2,6 +2,7 @@ export const REPOSITORIES_ROUTE = 'repositories';
 export const POPULAR_REPOSITORIES_ROUTE = 'popular';
 export const ADMIN_TASKS_ROUTE = 'admin-tasks';
 export const TEMPLATES_ROUTE = 'templates';
+export const TEMPLATES_DELETE_ROUTE = 'templates_delete';
```
Comment (xbhouse) [inline src/Routes/constants.ts:5]: can this be removed? i don't see it being used anywhere

**Comment by Andrewgdewar** [comment]
I pushed a small sizing/formatting change for the content of the modal. 
Unfortunately my linter was greedy touched all the things, so please do force push to reclaim your code 😅

**Review by Andrewgdewar — APPROVED** [review]
Ack !

**Pull request merged**